speechbrain/lobes/models/huggingface_transformers/llama.py

```diff
@@ -0,0 +1,133 @@
+"""This lobe enables the integration of huggingface pretrained LlaMA models.
+
+Authors
+ * Titouan Parcollet 2025
+ * Shucong Zhang 2025
+ * Pooneh Mousavi 2023
+"""
+
+from typing import Literal
+
+import torch
+from transformers import BitsAndBytesConfig
+
+from speechbrain.lobes.models.huggingface_transformers.huggingface import (
+    HFTransformersInterface,
+)
+from speechbrain.utils.logger import get_logger
+
+logger = get_logger(__name__)
+
+
+class LLaMA(HFTransformersInterface):
+    """This lobe enables the integration of HuggingFace pretrained LLaMA models.
+
+    The model can be finetuned entirely or coupled with SpeechBrain (and peft) adapters (see https://speechbrain.readthedocs.io/en/latest/tutorials/nn/neural-network-adapters.html)
+
+    Quantisation can be applied by passing a BitsAndBytesConfig which can be instantiated in a SpeechBrain yaml (or elsewhere.)
+
+    Transformer from HuggingFace needs to be installed:
+    https://huggingface.co/transformers/installation.html
+
+    Arguments
+    ---------
+    source : str
+        HuggingFace hub name: e.g "meta-llama/Llama-2-7b-chat-hf"
+    save_path : str
+        Path (dir) of the downloaded model.
+    bnb_config : transformers.BitsAndBytesConfig
+        BitsAndBytesConfig enabling quantisation of the model. If not specified, the model weights will be loaded with weight_precision_load dtype.
+    freeze : bool (default: false)
+        If True, the model is frozen. If False, the model will be trained
+        alongside with the rest of the pipeline.
+    pad_token : str (default: "PAD")
+        String representation of the padding token. This may change from one model to another.
+    padding_side : str (default: "right")
+        One of right or left.
+    weight_precision_load : torch.dtype (default: torch.bfloat16)
+        If no bnb_config is given, this parameter defines the loading type of the parameters of the model. This is useful to reduce memory footprint, but it does not change the compute dtype. For this just refer to mixed precision training in SpeechBrain.
+
+    Example
+    -------
+    >>> model_hub = "meta-llama/Llama-3.2-3B"
+    >>> save_path = "savedir"
+    >>> model = LLaMA(model_hub, save_path)
+    >>> tokens = torch.tensor([[1, 1]])
+    >>> attention_mask = torch.tensor([[1, 1]])
+    >>> outputs = model(tokens, attention_mask)
+    """
+
+    def __init__(
+        self,
+        source: str,
+        save_path: str,
+        bnb_config: BitsAndBytesConfig = None,
+        freeze: bool = False,
+        pad_token: str = "[PAD]",
+        padding_side: Literal["left", "right"] = "right",
+        weight_precision_load: torch.dtype = torch.bfloat16,
```

Comment:
That's why our HF interface is bad. The `from_pretrained` method gives you the possibility to override the precision of the backend through `torch_dtype`, and also the `attn_implementation`. E.g.:
```python
AutoModelForCausalLM.from_pretrained(
          params.backbone,
          torch_dtype=torch.bfloat16,
          attn_implementation="flash_attention_2"
)
```

I would therefore suggest to modify `huggingface.py` directly to allow both kwargs, and just define through the yaml those params (this class should also have a `*args` to accommodate this).

Reply:
makes sense

Reply:
@Adel-Moumen but wait, our interface already passes kwargs, there is nothing to change exept that I don't need to manage this on the llama lobe but just pass the dtype to the interface. right?

Reply:
you just need to add a kwargs to this interface, and then in the yaml to pass the `torch_dtype`. Also, you should maybe better document this behavior and especially the fact that you can dynamically modify the attn implemtantion through `attn_implementation`.

Reply:
@Adel-Moumen yes, this is what I ended up doing (see the CoVoST ast recipe). For the documentation, I guess you are right, i'll add a few words in the docstring.

Reply:
wait what happen to this?